src/main/scala/org/bitcoins/core/crypto/ECPrivateKey.scala

```diff
@@ -77,17 +82,54 @@ object ECPrivateKey extends Factory[ECPrivateKey] {
   }
 
   /**
-    * Takes in a base58 string and converts it into a private key
-    * @param base58
+    * Takes in a base58 string and converts it into a private key.
+    * Private keys starting with 'K', 'L', or 'c' correspond to compressed public keys.
+    * https://en.bitcoin.it/wiki/Wallet_import_format
+    *
+    * @param WIF Wallet Import Format. Encoded in Base58
+    * @return
+    */
+  def fromWIFToPrivateKey(WIF : String) : ECPrivateKey = {
+    val trimmedBytes = trimFunction(WIF)
+    val privateKeyBytesToHex = BitcoinSUtil.encodeHex(trimmedBytes)
+    ECPrivateKey(privateKeyBytesToHex)
+  }
+
+  /**
+    * Takes in WIF private key, as a string, and determines if it corresponds to a compressed public key
+    * If the private key corresponded to a compressed public key, also drop the last byte (it should be 0x01).
+    * If it corresponded to a compressed public key, the WIF string will have started with K or L instead of 5
+    * (or c instead of 9 on testnet).
+    *
+    * @param WIF
+    * @return
+    */
+  def isCompressed(WIF : String) : Boolean = {
+    val decoded = Base58.decode(WIF)
```

Comment:
This is relatively minor, but instead of calling `decode` right away inside of `isCompressed` you should just move this stuff inside of `isCompressed(bytes : Seq[Byte])` therefore you don't have to call the decode